src/Policy/OrmResolver.php

```diff
@@ -146,7 +159,13 @@ protected function findPolicy(string $class, string $name, string $namespace): m
             throw new MissingPolicyException([$class]);
         }
 
-        return new $policyClass();
+        if ($this->container && $this->container->has($policyClass)) {
+            $policy = $this->container->get($policyClass);
```

Comment:
Per the interface, `get()` can throw a variety of exceptions. Should this be handled by this method (i.e. fallback style) or leave it entirely to userland code?

Reply:
Well `get($policyClass)` won't throw any more since we checked the container with `has($policyClass)`.

This is the same procedure as we do when [parsing controller action arguments
](https://github.com/cakephp/cakephp/blob/5.x/src/Controller/ControllerFactory.php#L166-L169) just with the difference that the container could not be set in our case to be as BC as possible.

Reply:
We use the [league/container](https://github.com/thephpleague/container) as an implementation for the PSR-11 Container Interface. The only reason this exception could be thrown is [here](https://github.com/thephpleague/container/blob/4.x/src/Container.php#L195).

So technically, one could add a service provider which lies about providing a specific service.
But then I'd say its fine if we just let that exception bubble up to the top and not catch it because it's a configuration problem of the user (or a plugin), not something we should actively catch and handle for the user.